src/auth/apiheader/core/qgsauthapiheadermethod.cpp

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************
+    qgsauthapiheadermethod.cpp
+    --------------------------
+    begin                : October 2021
+    copyright            : (C) 2021 by Nyall Dawson
+    author               : Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsauthapiheadermethod.h"
+
+#include "qgsauthmanager.h"
+#include "qgslogger.h"
+#include "qgsapplication.h"
+
+#ifdef HAVE_GUI
+#include "qgsauthapiheaderedit.h"
+#endif
+
+#include <QNetworkProxy>
+#include <QMutexLocker>
+#include <QUuid>
+
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_KEY = QStringLiteral( "APIHeader" );
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_DESCRIPTION = QStringLiteral( "API Header" );
+const QString QgsAuthApiHeaderMethod::AUTH_METHOD_DISPLAY_DESCRIPTION = tr( "API Header" );
+
+QMap<QString, QgsAuthMethodConfig> QgsAuthApiHeaderMethod::sAuthConfigCache = QMap<QString, QgsAuthMethodConfig>();
+
+
+QgsAuthApiHeaderMethod::QgsAuthApiHeaderMethod()
+{
+  setVersion( 2 );
+  setExpansions( QgsAuthMethod::NetworkRequest );
+  setDataProviders( QStringList()
```

Comment:
Why restrict this to wms?

Reply:
The reason was the others hadn't been tested and its currently only required for WMS. Then again it's just a appending a HTTP header to the request, and as a result I've updated it to now include some others.